File: AddressBook/react-client/src/dtos/Timezone.ts
import Common from "../utility/Common";
import { CountryRes } from "./Country";
import { PagedReq } from "./PagedReq";

export interface TimezoneRes {
  timezoneId?: string;
  name?: string;
  cityName?: string;
  gmtOffset?: number;
  gmtOffsetName?: string;
  abbreviation?: string;

  countryId?: number;
  country?: CountryRes;
}

export class TimezoneReqEdit {
  name?: string = "";
  cityName?: string = "";
  gmtOffset?: number = 0;
  gmtOffsetName?: string = "";
  abbreviation?: string = "";

  countryId?: string = "";
  constructor(countryId?: string) {
    this.countryId = countryId;
  }
}

export class TimezoneReqSearch extends PagedReq {
  countryId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {countryId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.countryId = countryId;
  }
}

File: AddressBook/Dtos/Dtos/Timezone.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class TimezoneRes
    {
        public int TimezoneId { get; set; }
        public string? Name { get; set; }
        public string? CityName { get; set; }
        public int GmtOffset { get; set; }
        public string? GmtOffsetName { get; set; }
        public string? Abbreviation { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
        public CountryRes? Country { get; set; }
    }

    public class TimezoneReqEdit
    {
        [Required]
        public string? Name { get; set; }
        public string? CityName { get; set; }
        public int GmtOffset { get; set; }
        public string? GmtOffsetName { get; set; }
        public string? Abbreviation { get; set; }

        // Foreign keys
        public int? CountryId { get; set; }
    }

    public class TimezoneReqSearch : PagedReq
    {
        public int? CountryId { get; set; }
    }
}
